### PR TITLE
[Executor-benchmark] Make connected grps of txns fully conflicting grps

### PR DESCRIPTION
**Pull request by manudhundi**
Fully conflicting grps are more meaningful pattern for testing out the sharded
executor because they cannot be executed parallely (that is what we intend to benchmark).

Cmd:
cargo run -p aptos-executor-benchmark -- --block-size 100 --connected-tx-grps 5 --shuffle-connected-txns run-executor --main-signer-accounts 1000 --data-dir /tmp/some-db --checkpoint-dir /tmp/some-checkpoint --blocks 2

For a good random workload we would want 'num_accounts_per_grp' to be much greater than 'num_txns_per_grp'.
The command enforces 'num_signer_accounts' > '2 * num_txns_per_grp'.

### CODE REVIEW

**Comment by manudhundi** [comment]
#9164

**Comment by manudhundi** [comment]
#9371

File: execution/executor-benchmark/src/transaction_generator.rs [hunk 1]
```diff
@@ -451,91 +453,90 @@ impl TransactionGenerator {
...
+                            (index2, index1)
+                        }
+                    })
+                    .collect();
+                conflicting_indices
             })
             .collect();
-        transfer_indices.shuffle(rng);
+        if shuffle_indices {
+            println!("Shuffling conflicting txns*******************************");
```
Comment (sitalkedia) [inline execution/executor-benchmark/src/transaction_generator.rs:514]: Remove?

**Review by sitalkedia — APPROVED** [review]
LGTM

File: execution/executor-benchmark/src/transaction_generator.rs (hunk 1 above)
Comment (manudhundi) [reply execution/executor-benchmark/src/transaction_generator.rs:514]: Yes, will do

**Review by zjma — APPROVED** [review]

File: execution/executor-benchmark/src/main.rs [hunk 2]
```diff
@@ -128,6 +128,9 @@ struct Opt {
     #[clap(long, default_value_t = 0)]
     connected_tx_grps: usize,
 
+    #[clap(long)]
+    shuffle_connected_txns: bool,
```
[the remark this answers is not in the document]
Comment (manudhundi) [reply execution/executor-benchmark/src/main.rs:132]: Not shuffling is an interesting case where sharding does really well. Ideally, we want the sharding to do equally good in the case when it is shuffled. So, 'not shuffling' is a good benchmark to have.

Default value is false. There is some rust annotation issue if we specify default as positive. Can get around it by having a negated flag, but thought it is fine this way.

**Pull request merged**